neuland/digitizing/DigitizingTacQuila.cxx

```diff
@@ -12,175 +12,168 @@
+    bool Channel::HasFired()
+    {
+        if (!cachedFirstHitOverThresh.valid())
         {
-            if (!cachedFirstHitOverThresh.valid())
-            {
-                cachedFirstHitOverThresh.set(FindThresholdExceedingHit());
-                cachedQDC.invalidate();
-                cachedTDC.invalidate();
-                cachedEnergy.invalidate();
-            }
-            // If there is a hit that exceeded the threshold, the PMT fill fire
-            return cachedFirstHitOverThresh.get() != fPMTHits.end();
+            cachedFirstHitOverThresh.set(FindThresholdExceedingHit());
+            cachedQDC.invalidate();
+            cachedTDC.invalidate();
+            cachedEnergy.invalidate();
         }
+        // If there is a hit that exceeded the threshold, the PMT fill fire
+        return cachedFirstHitOverThresh.get() != fPMTHits.end();
+    }
 
-        Double_t Channel::GetQDC() const
+    Double_t Channel::GetQDC()
+    {
+        if (!cachedQDC.valid())
         {
-            if (!cachedQDC.valid())
-            {
-                cachedQDC.set(BuildQDC());
-            }
-            return cachedQDC;
+            cachedQDC.set(BuildQDC());
         }
+        return cachedQDC;
+    }
 
-        Double_t Channel::GetTDC() const
+    Double_t Channel::GetTDC()
+    {
+        if (!cachedTDC.valid())
         {
-            if (!cachedTDC.valid())
-            {
-                cachedTDC.set(BuildTDC());
-            }
-            return cachedTDC;
+            cachedTDC.set(BuildTDC());
         }
+        return cachedTDC;
+    }
 
-        Double_t Channel::GetEnergy() const
+    Double_t Channel::GetEnergy()
+    {
+        if (!cachedEnergy.valid())
         {
-            if (!cachedEnergy.valid())
-            {
-                cachedEnergy.set(BuildEnergy());
-            }
-            return cachedEnergy;
+            cachedEnergy.set(BuildEnergy());
         }
+        return cachedEnergy;
+    }
 
-        Double_t Channel::BuildQDC() const
+    Double_t Channel::BuildQDC()
+    {
+        if (HasFired())
         {
-            if (HasFired())
+            // @mheil: Should that be calculated using an exponential with the
+            // prior accumulated light / exponential decay or just all the light that arrives?
+            // Maximum pulse height or just sum over all light = charge?
+            Double_t light = 0.;
+            for (auto hit_it = cachedFirstHitOverThresh.get(); hit_it != fPMTHits.end(); hit_it++)
             {
-                // @mheil: Should that be calculated using an exponential with the
-                // prior accumulated light / exponential decay or just all the light that arrives?
-                // Maximum pulse height or just sum over all light = charge?
-                Double_t light = 0.;
-                for (auto hit_it = cachedFirstHitOverThresh.get(); hit_it != fPMTHits.end(); hit_it++)
+                const auto hit = *hit_it;
```

Comment:
Again, auto makes difficult the understanding.

Reply:
I only implemented tamex code. This was written by someone else long time ago. But I will only fix all the warnings relating to tacquila implementation.